Fix clang-tidy error in `dbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp`

## Enhancement
```
clang-tidy -p=/home/finder/projects/tiflash/build/ /home/finder/projects/tiflash/dbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp
/home/finder/projects/tiflash/contrib/boost/boost/config/detail/select_stdlib_config.hpp:26:14: error: 'cstddef' file not found [clang-diagnostic-error]
#    include <cstddef>
             ^
/home/finder/projects/tiflash/dbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp:47:23: error: method 'abandon' can be made static [readability-convert-member-functions-to-static,-warnings-as-errors]
void DeltaValueSpace::abandon(DMContext & context)
                      ^
/home/finder/projects/tiflash/dbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp:71:60: error: parameter 'head_column_files' is unused [misc-unused-parameters,-warnings-as-errors]
                                       const ColumnFiles & head_column_files,
                                                           ^~~~~~~~~~~~~~~~~
                                                            /*head_column_files*/
/home/finder/projects/tiflash/dbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp:102:23: error: method 'appendColumnFile' can be made static [readability-convert-member-functions-to-static,-warnings-as-errors]
bool DeltaValueSpace::appendColumnFile(DMContext & /*context*/, const ColumnFilePtr & column_file)
                      ^
static 
/home/finder/projects/tiflash/dbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp:102:87: error: parameter 'column_file' is unused [misc-unused-parameters,-warnings-as-errors]
bool DeltaValueSpace::appendColumnFile(DMContext & /*context*/, const ColumnFilePtr & column_file)
                                                                                      ^~~~~~~~~~~
                                                                                       /*column_file*/
/home/finder/projects/tiflash/dbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp:106:16: error: redundant boolean literal in conditional return statement [readability-simplify-boolean-expr,-warnings-as-errors]
        return false;
~~~~~~~~~~~~~~~^~~~~~
/home/finder/projects/tiflash/dbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp:112:23: error: method 'appendToCache' can be made static [readability-convert-member-functions-to-static,-warnings-as-errors]
bool DeltaValueSpace::appendToCache(DMContext & context, const Block & block, size_t offset, size_t limit)
                      ^
/home/finder/projects/tiflash/dbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp:116:16: error: redundant boolean literal in conditional return statement [readability-simplify-boolean-expr,-warnings-as-errors]
        return false;
~~~~~~~~~~~~~~~^~~~~~
/home/finder/projects/tiflash/dbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp:122:23: error: method 'appendDeleteRange' can be made static [readability-convert-member-functions-to-static,-warnings-as-errors]
bool DeltaValueSpace::appendDeleteRange(DMContext & /*context*/, const RowKeyRange & delete_range)
                      ^
/home/finder/projects/tiflash/dbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp:126:16: error: redundant boolean literal in conditional return statement [readability-simplify-boolean-expr,-warnings-as-errors]
        return false;
~~~~~~~~~~~~~~~^~~~~~
/home/finder/projects/tiflash/dbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp:132:23: error: method 'ingestColumnFiles' can be made static [readability-convert-member-functions-to-static,-warnings-as-errors]
bool DeltaValueSpace::ingestColumnFiles(DMContext & /*context*/, const RowKeyRange & range, const ColumnFiles & column_files, bool clear_data_in_range)
                      ^
static 
/home/finder/projects/tiflash/dbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp:132:113: error: parameter 'column_files' is unused [misc-unused-parameters,-warnings-as-errors]
bool DeltaValueSpace::ingestColumnFiles(DMContext & /*context*/, const RowKeyRange & range, const ColumnFiles & column_files, bool clear_data_in_range)
                                                                                                                ^~~~~~~~~~~~
                                                                                                                 /*column_files*/
/home/finder/projects/tiflash/dbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp:136:16: error: redundant boolean literal in conditional return statement [readability-simplify-boolean-expr,-warnings-as-errors]
        return false;
~~~~~~~~~~~~~~~^~~~~~
32812 warnings and 1 error generated.
Error while processing /home/finder/projects/tiflash/dbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp.
Suppressed 32819 warnings (32800 in non-user code, 19 NOLINT).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
12 warnings treated as errors
```

--- a/dbms/src/Storages/DeltaMerge/tests/gtest_dm_minmax_index.cpp
+++ b/dbms/src/Storages/DeltaMerge/tests/gtest_dm_minmax_index.cpp
@@ -323,44 +323,44 @@ RSOperatorPtr generateEqualOperator(MinMaxTestDatatype data_type, bool is_match)
     {
         if (is_match)
         {
-            return createEqual(attr("Date"), Field((String)Date_Match_DATA));
+            return createEqual(attr("Date"), Field(String(Date_Match_DATA)));
         }
         else
         {
-            return createEqual(attr("Date"), Field((String)Date_Smaller_DATA));
+            return createEqual(attr("Date"), Field(String(Date_Smaller_DATA)));
         }
     }
     case Test_Nullable_Date:
     {
         if (is_match)
         {
-            return createEqual(attr("Nullable(Date)"), Field((String)Date_Match_DATA));
+            return createEqual(attr("Nullable(Date)"), Field(String(Date_Match_DATA)));
         }
         else
         {
-            return createEqual(attr("Nullable(Date)"), Field((String)Date_Smaller_DATA));
+            return createEqual(attr("Nullable(Date)"), Field(String(Date_Smaller_DATA)));
         }
     }
     case Test_DateTime:
     {
         if (is_match)
         {
-            return createEqual(attr("DateTime"), Field((String)DateTime_Match_DATA));
+            return createEqual(attr("DateTime"), Field(String(DateTime_Match_DATA)));
         }
         else
         {
-            return createEqual(attr("DateTime"), Field((String)DateTime_Smaller_DATA));
+            return createEqual(attr("DateTime"), Field(String(DateTime_Smaller_DATA)));
         }
     }
     case Test_Nullable_DateTime:
     {
         if (is_match)
         {
-            return createEqual(attr("Nullable(DateTime)"), Field((String)DateTime_Match_DATA));
+            return createEqual(attr("Nullable(DateTime)"), Field(String(DateTime_Match_DATA)));
         }
         else
         {
-            return createEqual(attr("Nullable(DateTime)"), Field((String)DateTime_Smaller_DATA));
+            return createEqual(attr("Nullable(DateTime)"), Field(String(DateTime_Smaller_DATA)));
         }
     }
     case Test_MyDateTime:
@@ -442,44 +442,44 @@ RSOperatorPtr generateInOperator(MinMaxTestDatatype data_type, bool is_match)
     {
         if (is_match)
         {
-            return createIn(attr("Date"), {Field((String)Date_Match_DATA)});
+            return createIn(attr("Date"), {Field(String(Date_Match_DATA))});
         }
         else
         {
-            return createIn(attr("Date"), {Field((String)Date_Smaller_DATA)});
+            return createIn(attr("Date"), {Field(String(Date_Smaller_DATA))});
         }
     }
     case Test_Nullable_Date:
     {
         if (is_match)
         {
-            return createIn(attr("Nullable(Date)"), {Field((String)Date_Match_DATA)});
+            return createIn(attr("Nullable(Date)"), {Field(String(Date_Match_DATA))});
         }
         else
         {
-            return createIn(attr("Nullable(Date)"), {Field((String)Date_Smaller_DATA)});
+            return createIn(attr("Nullable(Date)"), {Field(String(Date_Smaller_DATA))});
         }
     }
     case Test_DateTime:
     {
         if (is_match)
         {
-            return createIn(attr("DateTime"), {Field((String)DateTime_Match_DATA)});
+            return createIn(attr("DateTime"), {Field(String(DateTime_Match_DATA))});
         }
         else
         {
-            return createIn(attr("DateTime"), {Field((String)DateTime_Smaller_DATA)});
+            return createIn(attr("DateTime"), {Field(String(DateTime_Smaller_DATA))});
         }
     }
     case Test_Nullable_DateTime:
     {
         if (is_match)
         {
-            return createIn(attr("Nullable(DateTime)"), {Field((String)DateTime_Match_DATA)});
+            return createIn(attr("Nullable(DateTime)"), {Field(String(DateTime_Match_DATA))});
         }
         else
         {
-            return createIn(attr("Nullable(DateTime)"), {Field((String)DateTime_Smaller_DATA)});
+            return createIn(attr("Nullable(DateTime)"), {Field(String(DateTime_Smaller_DATA))});
         }
     }
     case Test_MyDateTime:
@@ -561,44 +561,44 @@ RSOperatorPtr generateGreaterOperator(MinMaxTestDatatype data_type, bool is_matc
     {
         if (is_match)
         {
-            return createGreater(attr("Date"), Field((String)Date_Smaller_DATA), 0);
+            return createGreater(attr("Date"), Field(String(Date_Smaller_DATA)), 0);
         }
         else
         {
-            return createGreater(attr("Date"), Field((String)Date_Match_DATA), 0);
+            return createGreater(attr("Date"), Field(String(Date_Match_DATA)), 0);
         }
     }
     case Test_Nullable_Date:
     {
         if (is_match)
         {
-            return createGreater(attr("Nullable(Date)"), Field((String)Date_Smaller_DATA), 0);
+            return createGreater(attr("Nullable(Date)"), Field(String(Date_Smaller_DATA)), 0);
         }
         else
         {
-            return createGreater(attr("Nullable(Date)"), Field((String)Date_Match_DATA), 0);
+            return createGreater(attr("Nullable(Date)"), Field(String(Date_Match_DATA)), 0);
         }
     }
     case Test_DateTime:
     {
         if (is_match)
         {
-            return createGreater(attr("DateTime"), Field((String)DateTime_Smaller_DATA), 0);
+            return createGreater(attr("DateTime"), Field(String(DateTime_Smaller_DATA)), 0);
         }
         else
         {
-            return createGreater(attr("DateTime"), Field((String)DateTime_Match_DATA), 0);
+            return createGreater(attr("DateTime"), Field(String(DateTime_Match_DATA)), 0);
         }
     }
     case Test_Nullable_DateTime:
     {
         if (is_match)
         {
-            return createGreater(attr("Nullable(DateTime)"), Field((String)DateTime_Smaller_DATA), 0);
+            return createGreater(attr("Nullable(DateTime)"), Field(String(DateTime_Smaller_DATA)), 0);
         }
         else
         {
-            return createGreater(attr("Nullable(DateTime)"), Field((String)DateTime_Match_DATA), 0);
+            return createGreater(attr("Nullable(DateTime)"), Field(String(DateTime_Match_DATA)), 0);
         }
     }
     case Test_MyDateTime:
@@ -680,44 +680,44 @@ RSOperatorPtr generateGreaterEqualOperator(MinMaxTestDatatype data_type, bool is
     {
         if (is_match)
         {
-            return createGreaterEqual(attr("Date"), Field((String)Date_Smaller_DATA), 0);
+            return createGreaterEqual(attr("Date"), Field(String(Date_Smaller_DATA)), 0);
         }
         else
         {
-            return createGreaterEqual(attr("Date"), Field((String)Date_Greater_DATA), 0);
+            return createGreaterEqual(attr("Date"), Field(String(Date_Greater_DATA)), 0);
         }
     }
     case Test_Nullable_Date:
     {
         if (is_match)
         {
-            return createGreaterEqual(attr("Nullable(Date)"), Field((String)Date_Smaller_DATA), 0);
+            return createGreaterEqual(attr("Nullable(Date)"), Field(String(Date_Smaller_DATA)), 0);
         }
         else
         {
-            return createGreaterEqual(attr("Nullable(Date)"), Field((String)Date_Greater_DATA), 0);
+            return createGreaterEqual(attr("Nullable(Date)"), Field(String(Date_Greater_DATA)), 0);
         }
     }
     case Test_DateTime:
     {
         if (is_match)
         {
-            return createGreaterEqual(attr("DateTime"), Field((String)DateTime_Smaller_DATA), 0);
+            return createGreaterEqual(attr("DateTime"), Field(String(DateTime_Smaller_DATA)), 0);
         }
         else
         {
-            return createGreaterEqual(attr("DateTime"), Field((String)DateTime_Greater_DATA), 0);
+            return createGreaterEqual(attr("DateTime"), Field(String(DateTime_Greater_DATA)), 0);
         }
     }
     case Test_Nullable_DateTime:
     {
         if (is_match)
         {
-            return createGreaterEqual(attr("Nullable(DateTime)"), Field((String)DateTime_Smaller_DATA), 0);
+            return createGreaterEqual(attr("Nullable(DateTime)"), Field(String(DateTime_Smaller_DATA)), 0);
         }
         else
         {
-            return createGreaterEqual(attr("Nullable(DateTime)"), Field((String)DateTime_Greater_DATA), 0);
+            return createGreaterEqual(attr("Nullable(DateTime)"), Field(String(DateTime_Greater_DATA)), 0);
         }
     }
     case Test_MyDateTime:
@@ -799,44 +799,44 @@ RSOperatorPtr generateLessOperator(MinMaxTestDatatype data_type, bool is_match)
     {
         if (is_match)
         {
-            return createLess(attr("Date"), Field((String)Date_Greater_DATA), 0);
+            return createLess(attr("Date"), Field(String(Date_Greater_DATA)), 0);
         }
         else
         {
-            return createLess(attr("Date"), Field((String)Date_Match_DATA), 0);
+            return createLess(attr("Date"), Field(String(Date_Match_DATA)), 0);
         }
     }
     case Test_Nullable_Date:
     {
         if (is_match)
         {
-            return createLess(attr("Nullable(Date)"), Field((String)Date_Greater_DATA), 0);
+            return createLess(attr("Nullable(Date)"), Field(String(Date_Greater_DATA)), 0);
         }
         else
         {
-            return createLess(attr("Nullable(Date)"), Field((String)Date_Match_DATA), 0);
+            return createLess(attr("Nullable(Date)"), Field(String(Date_Match_DATA)), 0);
         }
     }
     case Test_DateTime:
     {
         if (is_match)
         {
-            return createLess(attr("DateTime"), Field((String)DateTime_Greater_DATA), 0);
+            return createLess(attr("DateTime"), Field(String(DateTime_Greater_DATA)), 0);
         }
         else
         {
-            return createLess(attr("DateTime"), Field((String)DateTime_Match_DATA), 0);
+            return createLess(attr("DateTime"), Field(String(DateTime_Match_DATA)), 0);
         }
     }
     case Test_Nullable_DateTime:
     {
         if (is_match)
         {
-            return createLess(attr("Nullable(DateTime)"), Field((String)DateTime_Greater_DATA), 0);
+            return createLess(attr("Nullable(DateTime)"), Field(String(DateTime_Greater_DATA)), 0);
         }
         else
         {
-            return createLess(attr("Nullable(DateTime)"), Field((String)DateTime_Match_DATA), 0);
+            return createLess(attr("Nullable(DateTime)"), Field(String(DateTime_Match_DATA)), 0);
         }
     }
     case Test_MyDateTime:
@@ -918,44 +918,44 @@ RSOperatorPtr generateLessEqualOperator(MinMaxTestDatatype data_type, bool is_ma
     {
         if (is_match)
         {
-            return createLessEqual(attr("Date"), Field((String)Date_Greater_DATA), 0);
+            return createLessEqual(attr("Date"), Field(String(Date_Greater_DATA)), 0);
         }
         else
         {
-            return createLessEqual(attr("Date"), Field((String)Date_Smaller_DATA), 0);
+            return createLessEqual(attr("Date"), Field(String(Date_Smaller_DATA)), 0);
         }
     }
     case Test_Nullable_Date:
     {
         if (is_match)
         {
-            return createLessEqual(attr("Nullable(Date)"), Field((String)Date_Greater_DATA), 0);
+            return createLessEqual(attr("Nullable(Date)"), Field(String(Date_Greater_DATA)), 0);
         }
         else
         {
-            return createLessEqual(attr("Nullable(Date)"), Field((String)Date_Smaller_DATA), 0);
+            return createLessEqual(attr("Nullable(Date)"), Field(String(Date_Smaller_DATA)), 0);
         }
     }
     case Test_DateTime:
     {
         if (is_match)
         {
-            return createLessEqual(attr("DateTime"), Field((String)DateTime_Greater_DATA), 0);
+            return createLessEqual(attr("DateTime"), Field(String(DateTime_Greater_DATA)), 0);
         }
         else
         {
-            return createLessEqual(attr("DateTime"), Field((String)DateTime_Smaller_DATA), 0);
+            return createLessEqual(attr("DateTime"), Field(String(DateTime_Smaller_DATA)), 0);
         }
     }
     case Test_Nullable_DateTime:
     {
         if (is_match)
         {
-            return createLessEqual(attr("Nullable(DateTime)"), Field((String)DateTime_Greater_DATA), 0);
+            return createLessEqual(attr("Nullable(DateTime)"), Field(String(DateTime_Greater_DATA)), 0);
         }
         else
         {
-            return createLessEqual(attr("Nullable(DateTime)"), Field((String)DateTime_Smaller_DATA), 0);
+            return createLessEqual(attr("Nullable(DateTime)"), Field(String(DateTime_Smaller_DATA)), 0);
         }
     }
     case Test_MyDateTime:
@@ -1051,7 +1051,7 @@ try
         {
             {
                 // not null
-                auto type_value_pair = generateTypeValue(MinMaxTestDatatype(datatype), false);
+                auto type_value_pair = generateTypeValue(static_cast<MinMaxTestDatatype>(datatype), false);
                 ASSERT_EQ(true, checkMatch(case_name, *context, type_value_pair.first, type_value_pair.second, generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_type), true)));
                 ASSERT_EQ(false, checkMatch(case_name, *context, type_value_pair.first, type_value_pair.second, generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_type), false)));
             }
@@ -1061,7 +1061,7 @@ try
                 {
                     continue;
                 }
-                auto type_value_pair = generateTypeValue(MinMaxTestDatatype(datatype), true);
+                auto type_value_pair = generateTypeValue(static_cast<MinMaxTestDatatype>(datatype), true);
                 ASSERT_EQ(true, checkMatch(case_name, *context, type_value_pair.first, type_value_pair.second, generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_type), true)));
                 ASSERT_EQ(true, checkMatch(case_name, *context, type_value_pair.first, type_value_pair.second, generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_type), false)));
             }
@@ -1071,7 +1071,7 @@ try
         {
             {
                 // not null
-                auto type_value_pair = generateTypeValue(MinMaxTestDatatype(datatype), false);
+                auto type_value_pair = generateTypeValue(static_cast<MinMaxTestDatatype>(datatype), false);
                 ASSERT_EQ(true, checkMatch(case_name, *context, type_value_pair.first, type_value_pair.second, generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_type), true)));
                 ASSERT_EQ(true, checkMatch(case_name, *context, type_value_pair.first, type_value_pair.second, generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_type), false)));
             }
@@ -1081,7 +1081,7 @@ try
                 {
                     continue;
                 }
-                auto type_value_pair = generateTypeValue(MinMaxTestDatatype(datatype), true);
+                auto type_value_pair = generateTypeValue(static_cast<MinMaxTestDatatype>(datatype), true);
                 ASSERT_EQ(true, checkMatch(case_name, *context, type_value_pair.first, type_value_pair.second, generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_type), true)));
                 ASSERT_EQ(true, checkMatch(case_name, *context, type_value_pair.first, type_value_pair.second, generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_type), false)));
             }
@@ -1101,7 +1101,7 @@ try
         {
             {
                 // not null
-                auto type_value_pair = generateTypeValue(MinMaxTestDatatype(datatype), false);
+                auto type_value_pair = generateTypeValue(static_cast<MinMaxTestDatatype>(datatype), false);
                 ASSERT_EQ(false, checkMatch(case_name, *context, type_value_pair.first, type_value_pair.second, createNot(generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_type), true))));
                 ASSERT_EQ(true, checkMatch(case_name, *context, type_value_pair.first, type_value_pair.second, createNot(generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_type), false))));
             }
@@ -1111,7 +1111,7 @@ try
                 {
                     continue;
                 }
-                auto type_value_pair = generateTypeValue(MinMaxTestDatatype(datatype), true);
+                auto type_value_pair = generateTypeValue(static_cast<MinMaxTestDatatype>(datatype), true);
                 ASSERT_EQ(true, checkMatch(case_name, *context, type_value_pair.first, type_value_pair.second, createNot(generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_type), true))));
                 ASSERT_EQ(true, checkMatch(case_name, *context, type_value_pair.first, type_value_pair.second, createNot(generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_type), false))));
             }
@@ -1133,7 +1133,7 @@ try
             {
                 {
                     // not null
-                    auto type_value_pair = generateTypeValue(MinMaxTestDatatype(datatype), false);
+                    auto type_value_pair = generateTypeValue(static_cast<MinMaxTestDatatype>(datatype), false);
                     auto left_rs_operator = generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_left_type), true);
                     auto right_rs_operator = generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_right_type), true);
                     ASSERT_EQ(true, checkMatch(case_name, *context, type_value_pair.first, type_value_pair.second, createAnd({left_rs_operator, right_rs_operator})));
@@ -1147,7 +1147,7 @@ try
                     {
                         continue;
                     }
-                    auto type_value_pair = generateTypeValue(MinMaxTestDatatype(datatype), true);
+                    auto type_value_pair = generateTypeValue(static_cast<MinMaxTestDatatype>(datatype), true);
                     auto left_rs_operator = generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_left_type), true);
                     auto right_rs_operator = generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_right_type), false);
                     ASSERT_EQ(true, checkMatch(case_name, *context, type_value_pair.first, type_value_pair.second, createAnd({left_rs_operator, right_rs_operator})));
@@ -1159,7 +1159,7 @@ try
             {
                 {
                     // not null
-                    auto type_value_pair = generateTypeValue(MinMaxTestDatatype(datatype), false);
+                    auto type_value_pair = generateTypeValue(static_cast<MinMaxTestDatatype>(datatype), false);
                     auto left_rs_operator = generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_left_type), true);
                     auto right_rs_operator = generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_right_type), false);
                     ASSERT_EQ(true, checkMatch(case_name, *context, type_value_pair.first, type_value_pair.second, createAnd({left_rs_operator, right_rs_operator})));
@@ -1171,7 +1171,7 @@ try
                     {
                         continue;
                     }
-                    auto type_value_pair = generateTypeValue(MinMaxTestDatatype(datatype), true);
+                    auto type_value_pair = generateTypeValue(static_cast<MinMaxTestDatatype>(datatype), true);
                     auto left_rs_operator = generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_left_type), true);
                     auto right_rs_operator = generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_right_type), false);
 
@@ -1197,7 +1197,7 @@ try
             {
                 {
                     // not null
-                    auto type_value_pair = generateTypeValue(MinMaxTestDatatype(datatype), false);
+                    auto type_value_pair = generateTypeValue(static_cast<MinMaxTestDatatype>(datatype), false);
                     auto left_rs_operator = generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_left_type), true);
                     auto right_rs_operator = generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_right_type), false);
                     ASSERT_EQ(true, checkMatch(case_name, *context, type_value_pair.first, type_value_pair.second, createOr({left_rs_operator, right_rs_operator})));
@@ -1211,7 +1211,7 @@ try
                     {
                         continue;
                     }
-                    auto type_value_pair = generateTypeValue(MinMaxTestDatatype(datatype), true);
+                    auto type_value_pair = generateTypeValue(static_cast<MinMaxTestDatatype>(datatype), true);
                     auto left_rs_operator = generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_left_type), true);
                     auto right_rs_operator = generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_right_type), false);
                     ASSERT_EQ(true, checkMatch(case_name, *context, type_value_pair.first, type_value_pair.second, createOr({left_rs_operator, right_rs_operator})));
@@ -1223,7 +1223,7 @@ try
             {
                 {
                     // not null
-                    auto type_value_pair = generateTypeValue(MinMaxTestDatatype(datatype), false);
+                    auto type_value_pair = generateTypeValue(static_cast<MinMaxTestDatatype>(datatype), false);
                     auto left_rs_operator = generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_left_type), false);
                     auto right_rs_operator = generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_right_type), false);
                     ASSERT_EQ(true, checkMatch(case_name, *context, type_value_pair.first, type_value_pair.second, createOr({left_rs_operator, right_rs_operator})));
@@ -1234,7 +1234,7 @@ try
                     {
                         continue;
                     }
-                    auto type_value_pair = generateTypeValue(MinMaxTestDatatype(datatype), true);
+                    auto type_value_pair = generateTypeValue(static_cast<MinMaxTestDatatype>(datatype), true);
                     auto left_rs_operator = generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_left_type), false);
                     auto right_rs_operator = generateRSOperator(static_cast<MinMaxTestDatatype>(datatype), static_cast<MinMaxTestOperator>(operater_right_type), false);
 

